Assets/Mirror/Runtime/Transport/Websocket/WebsocketTransport.cs

```diff
@@ -61,6 +64,21 @@ public override void ClientConnect(string host)
             }
         }
 
+        public override void ClientConnect(Uri uri)
+        {
+            if (uri.Scheme != Scheme && uri.Scheme != SecureScheme)
+                throw new ArgumentException($"Invalid url {uri}, use {Scheme}://host:port or {SecureScheme}://host:port instead", nameof(uri));
+
+            if (uri.IsDefaultPort)
+            {
+                UriBuilder uriBuilder = new UriBuilder(uri);
```

Comment:
can't we create a new Uri with (schema, address, port)? why UriBuilder?

Reply:
well,   at this point the only thing it is missing is the port,  the original uri already has the address and schema.   So just start from the original uri and add the default port

Reply:
Also theoretically,  one could also have a context and query string,  so I don't want to mess with the user provider uri any more than required